### PR TITLE
Check if the input is a non-empty string

### LINKED ISSUE #470: If pass empty string to dateparser.parse it tooks around 3-4 sec
If we pass empty string to parse method, it takes around 3-4 seconds to process it.
It should have some validations checks into it.

```
import dateparser
import time
st = time.time()
date = dateparser.parse('')
ed = time.time()
print('Date::', date)
print('%2.3f sec' % (ed - st))
```
```
Output:
>> Date:: 
>> 3.434 sec
```

### PR DESCRIPTION
**Pull request by rodp63**
I consider that checking if the input is a non-empty string is essential to start the parse process.

Fixes #470, closes #631

### CODE REVIEW

File: tests/test_input.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+from tests import BaseTestCase
+
+class InputTest(BaseTestCase):
+    def empty_input_test(self):
+        result = parse('')
+        self.assertFalse(result)
+
+    def whitespace_input_test(self):
+        result = parse(' \t\n\r\f\v')
+        self.assertFalse(result)
```
Comment (BurnzZ) [inline tests/test_input.py]: Could we have a test case to check out **Null Bytes**?

Comment (Rodp63) [reply tests/test_input.py]: Do you mean add a special condition? Something like:
`if (date_string == '\x00'): return`

Comment (BurnzZ) [reply tests/test_input.py]: Yes :)

Comment (Rodp63) [reply tests/test_input.py]: Got it! ;)

File: dateparser/__init__.py [hunk 2]
```diff
@@ -44,6 +45,10 @@ def parse(date_string, date_formats=None, languages=None, locales=None, region=N
     :rtype: :class:`datetime <datetime.datetime>`.
     :raises: ValueError - Unknown Language
     """
+
+    if not(re.compile('\S').search(date_string.replace('\x00',''))):
```
[the remark this answers is not in the document]
Comment (Rodp63) [reply dateparser/__init__.py]: Got it! You're right

**Comment by Gallaecio** [comment]
Re-running again. If the coverage data is to be believed, this is not working.

**Pull request closed without merging**